control-plane/api-gateway/binding/cleanup.go

```diff
@@ -0,0 +1,198 @@
+package binding
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/go-logr/logr"
+	"github.com/hashicorp/consul-k8s/control-plane/consul"
+	"github.com/hashicorp/consul/api"
+)
+
+const (
+	oldACLRoleName   = "managed-gateway-acl-role"
+	oldACLPolicyName = "api-gateway-token-policy"
+)
+
+var sleepTime = 60 * time.Second
```

Comment:
Pondering whether 60s here makes sense or if we should go with something like 10m intervals given that it's really just a cleanup job handling a long-term concern that isn't worth knocking on consul-server's door every 60s